src/gpu/intel/jit/gemm/generator/pieces/copy_plan.cpp

```diff
@@ -671,12 +671,18 @@ void CopyPlan::planTypeConversions()
         } else if (st == Type::ngen_f4_e3m0() && dt == DataType::hf) {
             planEmulatedE3M0ToHF(i);
             rerun = true;
-        } else if (dt == Type::ngen_f4_e2m1()) {
+        } else if (dt == Type::ngen_f4_e3m0() && st == DataType::hf) {
+            planEmulatedHFToE3M0(i);
+            rerun = true;
+        } else if (isFP4(dt)) {
             copyThrough(i, DataType::hf);
             rerun = true;
-        } else if (st == Type::ngen_f4_e2m1()) {
+        } else if (isFP4(st)) {
             copyThrough(i, DataType::hf);
             rerun = true;
+        } else if (one_of(st, DataType::ub, DataType::uw) && dt == DataType::u4) {
```

Comment:
Sign doesn't matter here, right?
```suggestion
        } else if ((isB(st) || isW(st)) && isInt4(dt)) {
```

Reply:
fixed, and updated function name.